stubs/fpdf2/fpdf/table.pyi

```diff
@@ -0,0 +1,67 @@
+from _typeshed import Incomplete
+from abc import abstractmethod
+from collections.abc import Iterable
+from dataclasses import dataclass
+from io import BytesIO
+from typing_extensions import Literal
+
+from PIL import Image
+
+from .drawing import DeviceGray, DeviceRGB
+from .enums import Align, TableBordersLayout, TableCellFillMode
+from .fonts import FontFace
+from .fpdf import FPDF
+
+DEFAULT_HEADINGS_STYLE: FontFace
+
+class Table:
+    rows: list[Row]
+
+    def __init__(
+        self,
+        fpdf: FPDF,
+        rows: Iterable[Incomplete] = (),
+        *,
+        align: str | Align = "CENTER",
+        borders_layout: str | TableBordersLayout = ...,
+        cell_fill_color: int | tuple[Incomplete, ...] | DeviceGray | DeviceRGB | None = None,
+        cell_fill_mode: str | TableCellFillMode = ...,
+        col_widths: int | tuple[int, ...] | None = None,
+        first_row_as_headings: bool = True,
+        headings_style: FontFace = ...,
+        line_height: int | None = None,
+        markdown=False,
+        text_align: str | Align = "JUSTIFY",
+        width: int | None = None,
+    ) -> None: ...
+    def row(self, cells: Iterable[str] = ()) -> Row: ...
+    def render(self) -> None: ...
+    def get_cell_border(self, i, j) -> str | Literal[0, 1]: ...
+
+class Row:
+    cells: list[Cell]
+    style: FontFace
+    def __init__(self, fpdf: FPDF) -> None: ...
+    @property
+    def cols_count(self) -> int: ...
+    def cell(
+        self,
+        text: str = "",
+        align: str | Align | None = None,
+        style: FontFace | None = None,
+        img: str | Image.Image | BytesIO | None = None,
+        img_fill_width: bool = False,
+        colspan: int = 1,
+    ) -> Cell: ...
+
+@dataclass
+class Cell:
+    text: str
+    align: str | Align | None
+    style: FontFace | None
+    img: str | None
+    img_fill_width: bool
+    colspan: int
+
+    @abstractmethod
+    def write(self, text, align: Incomplete | None = None): ...
```

Comment:
I'm not sure this is actually meant to be overridden on subclasses: it looks like the maintainers are raising `NotImplementedError` here because, well, they haven't implemented it yet: https://github.com/PyFPDF/fpdf2/blob/615db585a53dde9af5874fb0a58f33ee7f84aa2b/fpdf/table.py#L355

It's an unusual use of `NotImplementedError`, but it is a supported use of `NotImplementedError` according to the CPython docs! https://docs.python.org/3/library/exceptions.html#NotImplementedError

> In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to indicate that the real implementation still needs to be added.

```suggestion
    def write(self, text, align: Incomplete | None = None): ...
```